CMakePCHCompiler.cmake

```diff
@@ -72,40 +79,54 @@ function(target_precompiled_header) # target [...] header
 			endif()
 			if(MSVC)
 				# ensure pdb goes to the same location, otherwise we get C2859
-				file(TO_NATIVE_PATH
-					"${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/${target}.dir"
+				get_filename_component(
 					pdb_dir
+					"${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/${target}.dir"
+					ABSOLUTE
 					)
+				get_filename_component(win_pch "${target_dir}/${header}.pch" ABSOLUTE)
+				get_filename_component(win_header "${header}" ABSOLUTE)
 				# /Yc - create precompiled header
+				# /Fp - exact location for precompiled header
 				# /Fd - specify directory for pdb output
-				set(flags "/Yc /Fd${pdb_dir}\\")
+				set(flags "/Yc${win_header} /Fp${win_pch} /Fd${pdb_dir}")
+
+				set_source_files_properties(
+					${header}
+					PROPERTIES
+					LANGUAGE ${lang}
```

Comment:
Had to have the lang as CXX for VS for some reason or the file would not be built.